New setting compare-signatures-on-zone-freshness-check to disable DO flag for SOA checks (#11508)
PowerDNS/pdns: docs/settings.rst

service to 'simple' instead of 'notify' (refer to the systemd
documentation on how to modify unit-files).

.. _setting-compare-signatures-on-zone-freshness-check:

``compare-signatures-on-zone-freshness-check``
----------------------------------------------

.. versionadded:: 4.7.0

-  Boolean
-  Default: yes

> Turning this off will disable the DO flag for SOA queries during zone freshness checks of secondary zones

Reviewer: ```suggestion
Turning this off will avoid sending the DO flag for SOA queries during zone freshness checks of secondary zones
```

> to workaround truncated SOA responses. It will also disable signature comparison which are used to detect

Reviewer: ```suggestion
to work around truncated SOA responses. It will also disable signature comparison which is used to detect
```

> to work around truncated SOA responses. It will also disable signature comparison which is used to detect
> signature changes even when the serial was not increased. Hence, disable this setting only if the Primary

Reviewer: ```suggestion
to work around truncated SOA responses. It will also disable signature comparison which is used to detect
signature changes without changes to the serial. Hence, disable this setting only if the Primary
```

> name server always increases the serial on signature changes.

Reviewer: ```suggestion
name server always increases the serial when it updates signatures.
```

service to 'simple' instead of 'notify' (refer to the systemd
documentation on how to modify unit-files).

.. _setting-secondary-check-signature-freshness:

``secondary-check-signature-freshness``
---------------------------------------

.. versionadded:: 4.7.0

-  Boolean
-  Default: yes

Enabled by default, freshness checks for secondary zones will set the DO flag on SOA queries. PowerDNS
can detect (signature) changes on the primary server without serial number bumps using the DNSSEC
signatures in the SOA response.

In some problematic scenarios, primary servers send truncated SOA responses. As a workaround, this setting
can be turned off, and the DO flag as well as the signature checking will be disabled. To avoid additional

> drift, primary servers then must always increase the zone serial on signature changes.

Reviewer: ```suggestion
drift, primary servers must then always increase the zone serial on signature changes.
```

service to 'simple' instead of 'notify' (refer to the systemd
documentation on how to modify unit-files).

.. _setting-compare-signatures-on-zone-freshness-check:

``compare-signatures-on-zone-freshness-check``
----------------------------------------------

.. versionadded:: 4.7.0

-  Boolean
-  Default: yes

Enabled by default, freshness checks for secondary zones will set the DO flag on SOA queries. PowerDNS

> uses the DNSSEC signatures in the SOA response to detect (signature) changes on the primary server, when
> the serial number was not increased.

Reviewer: ```suggestion
can detect (signature) changes on the primary server without serial number bumps using the DNSSEC signatures in the SOA response.
```

Reviewer 2: thanks, committed!
